### PR TITLE
Migrate mail tests

### PR DESCRIPTION
**Pull request by trasher**

### CODE REVIEW

**Comment by trasher** [comment]
Currently fails on a test using a provider... I wonder if the `eval()` used in this provider is not the cause; I cannot test locally.

File: phpunit/imap/MailCollectorTest.php [hunk 1]
```diff
@@ -1246,22 +1248,21 @@ public function testGetAdditionnalMailServerProtocols(
...
-            ->withType(E_USER_WARNING)
-            ->withMessage($expected_warning)
-            ->exists();
-        } else {
-            $getProtocol();
-        }
 
-       // Get storage
+        $getProtocol();
+        /*if ($expected_warning !== null) {
```
Comment (trasher) [inline phpunit/imap/MailCollectorTest.php:1258]: It's strange I had to comment those ones :/

**Comment by cedric-anne** [comment]
> > Currently fails on a test using a provider... I wonder if the `eval()` used in this provider is not the cause; I cannot test locally.
> 
> I've replaced `eval` with real classes from files, tests still fails... I guess this is another gap between atoum and phpunit, but I've no idea what's wrong here.

The plugin was not marked as activated/loaded, and therefore the corresponding hooks were not executed due to https://github.com/glpi-project/glpi/blob/956702eabbe456e9f1b0d1e12ddd46fa4e3b195f/src/Plugin.php#L1757-L1759

**Review by cedric-anne — APPROVED** [review]

**Comment by trasher** [comment]
> The plugin was not marked as activated/loaded, and therefore the corresponding hooks were not executed due to

OK, strange it did not break before then :)

**Comment by cedric-anne** [comment]
> > The plugin was not marked as activated/loaded, and therefore the corresponding hooks were not executed due to
> 
> OK, strange it did not break before then :)

Plugins loading state is stored in static variables, this is probably the problem here.

**Pull request merged**